### PR TITLE
Sign Post Check Enhancement Review

### PR DESCRIPTION
**Pull request by Bentleysb**
This enhancement changes default configurable values, removes the `motorway_junction` test from on ramps, reworks the `findFirstRampEdge` method, and enables this check.  

The new default values create less errors from overlapping values.

The `motorway_junction` test was removed from on ramps because they do not require a node with that tag: [OSM wiki: motorway_junction](https://wiki.openstreetmap.org/wiki/Tag:highway=motorway_junction)

`findFirstRampEdge` no longer uses heading to determine connection. It now uses a recursive search for arterial roads based on the configurable filters and minimum arterial classification. 

This check was disabled via the configuration. The configuration.json in this PR re-enables it.

### CODE REVIEW

File: src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SignPostCheck.java [hunk 1]
```diff
@@ -195,53 +199,50 @@ public boolean validCheckForObject(final AtlasObject object)
...
-            }
-
-            // Break if given edge has no heading
-            final Optional<Heading> finalHeading = precedingEdge.asPolyLine().finalHeading();
-            if (!finalHeading.isPresent() || initialHeading.get().asPositiveAngle()
-                    .difference(finalHeading.get().asPositiveAngle())
-                    .isGreaterThan(this.rampAngleDifference))
-            {
-                break;
+                // Check each in edge
```
Comment (danielduhh) [inline src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SignPostCheck.java]: nit: inEdge

File: src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SignPostCheck.java [hunk 2]
```diff
@@ -195,53 +199,50 @@ public boolean validCheckForObject(final AtlasObject object)
...
+                            .orElse(HighwayTag.NO).isMoreImportantThanOrEqualTo(arterialMinimum))
+                    {
+                        endEdges.clear();
+                        endEdges.add(finalEdge);
+                        return endEdges;
+                    }
+                    // Recurse through inEdges if this edge is not the other side of a bidirectional
+                    // way
+                    else if (nextEdge.highwayTag().isIdenticalClassification(finalEdge.highwayTag())
+                            && !(finalEdge.getMasterEdgeIdentifier() == nextEdge
```
Comment (danielduhh) [inline src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SignPostCheck.java]: Is there a reason why you wrapped the identifier boolean in a paren? Can you just 

```java
finalEdge.getMasterEdgeIdentifier() != nextEdge.getMasterEdgeIdentifier()
```

File: src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SignPostCheck.java [hunk 3]
```diff
@@ -195,53 +199,50 @@ public boolean validCheckForObject(final AtlasObject object)
      *            {@link Edge} that is the last/final piece of the ramp
      * @return {@link Edge} that possibly is the first {@link Edge} forming the ramp
      */
-    private Edge findFirstRampEdge(final Edge finalEdge)
+    private List<Edge> findFirstRampEdge(final Edge finalEdge, final long maxEdgeCount)
     {
-        int count = 0;
+        final int count = 0;
```
Comment (danielduhh) [inline src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SignPostCheck.java]: Is this being incremented anywhere?

File: src/test/java/org/openstreetmap/atlas/checks/validation/linear/edges/SignPostCheckTest.java [hunk 4]
```diff
@@ -26,6 +24,10 @@
                             + "\"SignPostCheck.ramp.differentiator.tag\": \"diff\","
                             + "\"SignPostCheck.ramp.angle.difference.degrees\": 90.0" + "}"));
 
+    private static SignPostCheck CHECK2 = new SignPostCheck(
```
Comment (danielduhh) [inline src/test/java/org/openstreetmap/atlas/checks/validation/linear/edges/SignPostCheckTest.java:27]: Whats the deal with the `SignPostCheck.ramp.differentiator.tag` in the `CHECK` variable?

**Review by danielduhh — CHANGES_REQUESTED** [review]
Nice work B - had some minor questions/requests

File: src/test/java/org/openstreetmap/atlas/checks/validation/linear/edges/SignPostCheckTest.java (hunk 4 above)
Comment (Bentleysb) [reply src/test/java/org/openstreetmap/atlas/checks/validation/linear/edges/SignPostCheckTest.java:27]: `SignPostCheck.ramp.differentiator.tag` is used to detect if an edge is likely a ramp. The original unit tests check that it works by using a made up tag.

**Comment by danielduhh** [comment]
failing on gradle build

```bash
* What went wrong:
Execution failed for task ':spotlessJava'.
> The following files had format violations:
      src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SignPostCheck.java
          @@ -230,7 +230,9 @@
           ····················}
           ····················//·Recurse·through·inEdges·if·this·edge·is·not·the·other·side·of·a·bidirectional
           ····················//·way
          -····················else·if·(nextEdge.highwayTag().isIdenticalClassification(finalEdge.highwayTag())·&&·finalEdge.getMasterEdgeIdentifier()·!=·nextEdge.getMasterEdgeIdentifier())
          +····················else·if·(nextEdge.highwayTag().isIdenticalClassification(finalEdge.highwayTag())
          +····························&&·finalEdge.getMasterEdgeIdentifier()·!=·nextEdge
          +····································.getMasterEdgeIdentifier())
           ····················{
           ························endEdges.addAll(findFirstRampEdge(nextEdge,·maxEdgeCount·-·1));
           ····················}
  Run 'gradlew spotlessApply' to fix these violations.
```

**Comment by Bentleysb** [comment]
No build problem on my end. Tried on a fresh clone of the fork/branch.

**Review by danielduhh — APPROVED** [review]

**Pull request closed without merging**